GithubRepo/Services/GithubFetcher.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  API.swift
+//  GithubRepo
+//
+//  Created by Marcos Alves on 02/10/21.
+//
+
+import Foundation
+
+struct GithubFetcherConstants {
+    static let kUrl = "https://api.github.com/search/repositories?q="
+}
+
+
+struct GithubFetcher {
+    
+    enum GithubFetcherError: Error {
+        case invalidURL
+        case missingData
+    }
+    
+    static func fetchRepositories(query: String, completion: @escaping(Result<SearchRepoResponse, Error>) -> Void) {
```

Comment:
tip: you can return the `task` object here to be able to cancel the request in the future.